### PR TITLE
feat: add scripts/install.ps1

### PR DESCRIPTION
**Pull request by amdprophet**
Implements an install script in PowerShell for Windows that can detect OS type, CPU architecture, installed otelcol-sumo version, latest otelcol-sumo version, and can accept the following params:
* `-Version <version>`
* `-InstallationToken <installation_token>` (defaults to the `SUMOLOGIC_INSTALLATION_TOKEN` envvar)
* `-Tags <tags_as_hashtable>` (e.g. `-Tags @{ tag1 = "foo"; tag2 = "bar" }`)

The following command can be used to download & install after replacing `replaceme` with your installation token:

```powershell
Set-ExecutionPolicy RemoteSigned -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; $uri = "https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/feat/install-ps1/scripts/install.ps1"; $path="${env:TEMP}\install.ps1"; (New-Object System.Net.WebClient).DownloadFile($uri, $path); . $path -InstallationToken replaceme -Tags @{tag3 = "foo"; tag4 = "bar"}
```

### CODE REVIEW

File: scripts/install.ps1 [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+    Write-Host "Getting installed version..."
+    $installedVersion = Get-InstalledVersion
+    $installedVersionStr = "none"
+    if ($installedVersion -ne $null) {
+        $installedVersionStr = $installedVersion
+    }
+    Write-Host "Installed version:`t${installedVersionStr}"
+
+    # Get versions, but ignore errors are we fallback to other methods later
```
Comment (sumo-drosiek) [inline scripts/install.ps1:397]:
```suggestion
    # Get versions, but ignore errors as we fallback to other methods later
```

File: scripts/install.ps1 [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+    $msiLanguage = "en-US"
+    $msiFileName = "otelcol-sumo_${productVersion}_${msiLanguage}.${archName}.msi"
+    $msiUri = "https://github.com/SumoLogic/sumologic-otel-collector/releases/download/"
+    $msiUri += "v${Version}/${msiFileName}"
+    $msiPath = "${env:TEMP}\${msiFileName}"
+    Get-BinaryFromUri $msiUri -Path $msiPath -HttpClient $httpClient
+
+    # Install MSI
+    [string[]] $msiProperties = @()
+    $msiProperties += "INSTALLATIONTOKEN=${InstallationToken}"
```
Comment (sumo-drosiek) [inline scripts/install.ps1:451]: Does it work for `v0.72.0` release? As I changed the name recently I would expect it to work from `v0.73.0` :thinking:

Comment (amdprophet) [reply scripts/install.ps1:451]: Yes, it worked when I was testing it and it looks like it was included in `v0.72.0` https://github.com/SumoLogic/sumologic-otel-collector/blob/v0.72.0-sumo-0/packaging/msi/wix/package.wxs

Comment (sumo-drosiek) [reply scripts/install.ps1:451]: eh, seems that I unnecessary added support for `INSTALLTOKEN` again :roll_eyes:

**Pull request merged**